object_store/src/throttle.rs

```diff
@@ -0,0 +1,500 @@
+
+        store.wait_get_per_call = WAIT_TIME;
+        store.wait_get_per_byte = WAIT_TIME;
+        assert_bounds!(measure_get(&store, Some(2)).await, 3);
+    }
+
+    #[tokio::test]
+    async fn list_test() {
+        let inner = InMemory::new();
+        let mut store = ThrottledStore::new(inner);
+
+        assert_bounds!(measure_list(&store, 0).await, 0);
+        assert_bounds!(measure_list(&store, 10).await, 0);
+
+        store.wait_list_per_call = WAIT_TIME;
+        assert_bounds!(measure_list(&store, 0).await, 1);
+        assert_bounds!(measure_list(&store, 10).await, 1);
+
+        store.wait_list_per_call = ZERO;
+        store.wait_list_per_entry = WAIT_TIME;
+        assert_bounds!(measure_list(&store, 2).await, 2);
+
+        store.wait_list_per_call = WAIT_TIME;
+        store.wait_list_per_entry = WAIT_TIME;
+        assert_bounds!(measure_list(&store, 2).await, 3);
+    }
+
+    #[tokio::test]
+    async fn list_with_delimiter_test() {
+        let inner = InMemory::new();
+        let mut store = ThrottledStore::new(inner);
+
+        assert_bounds!(measure_list_with_delimiter(&store, 0).await, 0);
+        assert_bounds!(measure_list_with_delimiter(&store, 10).await, 0);
+
+        store.wait_list_with_delimiter_per_call = WAIT_TIME;
+        assert_bounds!(measure_list_with_delimiter(&store, 0).await, 1);
+        assert_bounds!(measure_list_with_delimiter(&store, 10).await, 1);
+
+        store.wait_list_with_delimiter_per_call = ZERO;
+        store.wait_list_with_delimiter_per_entry = WAIT_TIME;
+        assert_bounds!(measure_list_with_delimiter(&store, 2).await, 2);
+
+        store.wait_list_with_delimiter_per_call = WAIT_TIME;
+        store.wait_list_with_delimiter_per_entry = WAIT_TIME;
+        assert_bounds!(measure_list_with_delimiter(&store, 2).await, 3);
+    }
+
+    #[tokio::test]
+    async fn put_test() {
+        let inner = InMemory::new();
+        let mut store = ThrottledStore::new(inner);
+
+        assert_bounds!(measure_put(&store, 0).await, 0);
+        assert_bounds!(measure_put(&store, 10).await, 0);
+
+        store.wait_put_per_call = WAIT_TIME;
+        assert_bounds!(measure_put(&store, 0).await, 1);
+        assert_bounds!(measure_put(&store, 10).await, 1);
+
+        store.wait_put_per_call = ZERO;
+        store.wait_put_per_byte = WAIT_TIME;
+        assert_bounds!(measure_put(&store, 2).await, 2);
+
+        store.wait_put_per_call = WAIT_TIME;
+        store.wait_put_per_byte = WAIT_TIME;
+        assert_bounds!(measure_put(&store, 2).await, 3);
+    }
+
+    async fn place_test_object(
+        store: &ThrottledStore<InMemory>,
+        n_bytes: Option<usize>,
+    ) -> <ThrottledStore<InMemory> as ObjectStoreApi>::Path {
+        let mut path = store.new_path();
+        path.set_file_name("foo");
+
+        if let Some(n_bytes) = n_bytes {
+            let data = Bytes::from_iter((0..n_bytes).map(|_| 1u8));
+            let stream_data = std::io::Result::Ok(data.clone());
```

Comment:
I don't think this `clone` is needed here, since `data` isn't used anywhere else?
```suggestion
            let stream_data = std::io::Result::Ok(data);
```